### DISCUSSION TITLE
How to introduce Z3 in Visual Studio

### DISCUSSION THREAD

**Discussion by xebelleDerrel** [Z3Prover/z3#7177] [Q&A]
I'd like to introduce Z3 into my own Visual Studio empty project, but it seems that only the way to link static libraries is provided in Visual Studio, but Z3 doesn't seem to support it, so can I add some source files from Z3 for manual compilation, or do you think there is any workable solution

**Comment by wintersteiger** [comment]
To use Z3 as a library with Visual C, you need to build it with Visual C. The Z3 releases come with a pre-built version (look for the x64-win zip). This contains `libz3.lib` and `libz3.dll`. As usual for Visual C, you pass the `.lib` file to your linker and the `.dll` will be loaded at runtime. All of this is exactly the same as for any other Visual C library, so StackOverflow for instance, can answer all your questions.

There are also ways to build static Z3 libraries, but they are not included in the release packages.
